### PR TITLE
Build image with Qt6

### PR DESCRIPTION
**Pull request by saw-jan**
With this PR, the image will have QT6 (`6.5.1`) required for the desktop client version 5.0

### CODE REVIEW

**Comment by saw-jan** [comment]
@fmoc Could you review this and merge if it's ok. Then we can make ctest pipeline green in drone-ci

**Comment by saw-jan** [comment]
I built and checked the image in https://github.com/owncloud/client/pull/10971

File: fedora/Dockerfile.amd64 [hunk 1]
```diff
@@ -58,19 +58,21 @@ RUN yum install -y \
...
+    qt6-qtbase-devel \
+    qt6-qttools-devel \
+    qtkeychain-qt6-devel \
     libcloudproviders-devel \
-    qtkeychain-qt5-devel \
     kf5-kio-devel \
     && yum autoremove -y \
     && yum clean all
 
+ENV CMAKE_PREFIX_PATH=/usr/lib64/cmake/Qt6
```
Comment (fmoc) [inline fedora/Dockerfile.amd64:69]: Should not be needed, but won't hurt either.

File: fedora/Dockerfile.amd64 [hunk 2]
```diff
@@ -80,7 +82,7 @@ RUN git clone https://github.com/owncloud/libre-graph-api-cpp-qt-client && \
     cd libre-graph-api-cpp-qt-client/client && \
     mkdir build && \
     cd build && \
-    cmake .. -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_INSTALL_PREFIX=/usr -G Ninja && \
+    cmake .. -DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH} -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_INSTALL_PREFIX=/usr -G Ninja && \
```
Comment (fmoc) [inline fedora/Dockerfile.amd64]:
```suggestion
    cmake .. -DCMAKE_PREFIX_PATH="${CMAKE_PREFIX_PATH}" -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_INSTALL_PREFIX=/usr -G Ninja && \
```

**Review by fmoc — CHANGES_REQUESTED** [review]

**Review by fmoc — APPROVED** [review]

**Pull request merged**